### PR TITLE
Add basic support for WebExtension dictionaries

### PR DESCRIPTION
**Pull request by diox**
Fix https://github.com/mozilla/addons-linter/issues/1984

### CODE REVIEW

File: src/parsers/manifestjson.js [hunk 1]
```diff
@@ -185,6 +189,20 @@ export default class ManifestJSONParser extends JSONParser {
       });
     }
 
+    log.info(this.parsedJSON.dictionaries);
```
Comment (diox) [inline src/parsers/manifestjson.js]: I should remove this.

**Comment by diox** [comment]
@EnTeQuAk can you help me figure out why the tests are not passing ?

**Comment by EnTeQuAk** [comment]
Tests are failing because of eslint failures:

```
/home/travis/build/mozilla/addons-linter/src/messages/manifestjson.js
  170:53  error  'type' is defined but never used  no-unused-vars
/home/travis/build/mozilla/addons-linter/src/parsers/manifestjson.js
  196:13  error  'filepath' is never reassigned. Use 'const' instead  prefer-const
  199:1   error  Expected indentation of 8 spaces but found 10        indent
  200:1   error  Expected indentation of 8 spaces but found 10        indent
  201:1   error  Expected indentation of 10 spaces but found 14       indent
  202:1   error  Expected indentation of 10 spaces but found 14       indent
/home/travis/build/mozilla/addons-linter/tests/unit/parsers/test.manifestjson.js
  1076:68  error  A space is required before '}'  object-curly-spacing
  1104:47  error  A space is required before '}'  object-curly-spacing
  1118:85  error  A space is required before '}'  object-curly-spacing
```

You can run this locally via `npm run eslint`

File: src/schema/validator.js [hunk 2]
```diff
@@ -81,6 +81,18 @@ export const validateLangPack = (...args) => {
...
+  ...schemaObject,
+  id: 'dictionary-manifest',
+  $ref: '#/types/WebExtensionDictionaryManifest',
+});
+
+export const validateDictionary = (...args) => {
+  const isValid = _validateDictionary(...args);
+  validateDictionary.errors = filterErrors(_validateDictionary.errors);
+  return isValid;
+};
```
Comment (EnTeQuAk) [inline src/schema/validator.js]: Similarly to static themes I'd guess that dictionaries should never contain anything in the manifest from regular WebExtensions, e.g `content_script` or others should error.

We achieve this by injecting `additionalProperties: false` as you can see here: https://github.com/mozilla/addons-linter/pull/2121/files#diff-949170b1ae8ed6356b1a78fd751d87a0R111

But that can very easily be a follow-up with more validation happening for `.aff` and `.dic` files too.

**Comment by diox** [comment]
Somehow totally missed the eslint failures in the travis logs... will fix.

File: src/schema/validator.js (hunk 2 above)
Comment (diox) [reply src/schema/validator.js]: Yeah I was planning to do this in https://github.com/mozilla/addons-linter/issues/2055

**Pull request merged**